sqlglot/optimizer/qualify_columns.py

```diff
@@ -65,7 +66,7 @@ def validate_qualify_columns(expression):
     for scope in traverse_scope(expression):
         if isinstance(scope.expression, exp.Select):
             unqualified_columns.extend(scope.unqualified_columns)
-            if scope.external_columns and not scope.is_correlated_subquery:
+            if scope.external_columns and not scope.is_correlated_subquery and not scope.pivots:
```

Comment:
This check was needed because the columns qualified with the pivot's alias are treated as external by the scope and so we'd hit the "column could not be resolved for table ..." error.